releases/release-5.0.4.md

```diff
@@ -0,0 +1,197 @@
+        - 移除 `file sorter` 文件排序器 [#2326](https://github.com/pingcap/ticdc/pull/2326)
+        - 总是从 TiKV 获取行变更的历史值 (old value)，输出会根据 `enable-old-value` 配置进行调整 [#2301](https://github.com/pingcap/ticdc/issues/2301)
+        - 优化连接 PD 时缺少证书情况下的报错提示 [#1973](https://github.com/pingcap/ticdc/issues/1973)
+        - 在高并发下减少 workerpool 中创建的 goroutine 数量 [#2211](https://github.com/pingcap/ticdc/issues/2211)
+        - 为所有 KV 客户端创建全局共享的 gRPC 连接池 [#2533](https://github.com/pingcap/ticdc/pull/2533)
+
+## Bug 修复
+
++ TiDB
+
+    - 修复了当查询分区表且分区键带有 `IS NULL` 条件时，TiDB 可能 panic 的问题 [#23802](https://github.com/pingcap/tidb/issues/23802)
+    - 修复了 `FLOAT64` 类型的溢出检查与 MySQL 不一致的问题 [#23897](https://github.com/pingcap/tidb/issues/23897)
+    - 修复了 `case when` 表达式的字符集和排序规则错误的问题 [#26662](https://github.com/pingcap/tidb/issues/26662)
+    - 修复了提交悲观事务可能会导致写冲突的问题 [#25964](https://github.com/pingcap/tidb/issues/25964)
+    - 修复了在悲观事务中索引键值可能会被重复提交的问题 [#26359](https://github.com/pingcap/tidb/issues/26359) [#10600](https://github.com/tikv/tikv/pull/10600)
+    - 修复了清除 Async Commit 锁时可能会导致 TiDB panic 的问题 [#25778](https://github.com/pingcap/tidb/issues/25778)
+    - 修复了使用 `INDEX MERGE` 时可能找不到列的问题 [#25045](https://github.com/pingcap/tidb/issues/25045)
+    - 修复了使用 `ALTER USER REQUIRE SSL` 会清空用户的 `authentication_string` 的问题 [#25225](https://github.com/pingcap/tidb/issues/25225)
+    - 修复了新集群上系统变量 `tidb_gc_scan_lock_mode` 的值显示为 "PHYSICAL" 而实际是 "LEGACY" 的问题 [#25100](https://github.com/pingcap/tidb/issues/25100)
+    - 修复了系统表 `TIKV_REGION_PEERS` 没有正确显示 `DOWN` 状态的问题 [#24879](https://github.com/pingcap/tidb/issues/24879)
+    - 修复了使用 HTTP API 时导致内存泄漏的问题 [#24649](https://github.com/pingcap/tidb/pull/24649)
+    - 修复了视图不支持 `DEFINER` 的问题 [#24414](https://github.com/pingcap/tidb/issues/24414)
+    - 修复了 `tidb-server --help` 退出状态是 `2` 的问题 [#24046](https://github.com/pingcap/tidb/issues/24046)
+    - 修复了设置全局系统变量 `dml_batch_size` 不生效的问题 [#24709](https://github.com/pingcap/tidb/issues/24709)
+    - 修复了同时使用 `read_from_storage` 和分区表会报错的问题 [#20372](https://github.com/pingcap/tidb/issues/20372)
+    - 修复了 TiDB 在执行投影算子时会 Panic 的问题 [#24264](https://github.com/pingcap/tidb/issues/24264)
+    - 修复了统计信息可能导致查询 Panic 的问题 [#24061](https://github.com/pingcap/tidb/pull/24061)
+    - 修复了在 `BIT` 类型的列上使用 `approx_percentile` 函数可能 Panic 的问题 [#23662](https://github.com/pingcap/tidb/issues/23662)
+    - 修复了 Grafana 上 `Coprocessor Cache` 面板的数据显示不正确的问题 [#26338](https://github.com/pingcap/tidb/issues/26338)
+    - 修复了并发 truncate 同一个分区会导致 DDL 语句执行卡住的问题 [#26229](https://github.com/pingcap/tidb/issues/26229)
+    - 修复当会话变量用作 `GROUP BY` 项时查询结果出错的问题 [#27106](https://github.com/pingcap/tidb/issues/27106)
+    - 修复连接表时 `VARCHAR` 类型与时间戳之间错误的隐式转换 [#25902](https://github.com/pingcap/tidb/issues/25902)
+    - 修复相关子查询语句中的错误结果 [#27233](https://github.com/pingcap/tidb/issues/27233)
+
++ TiKV
+
+    ?- 修复当有快照文件未被 GC 回收时，快照在 GC 的过程中可能遗留 GC 快照文件的问题 [#10813](https://github.com/tikv/tikv/issues/10813)
```

Comment:
- Fix the issue that snapshot GC might miss GC snapshot files when there is a snapshot file failed to be garbage-collected [#10813](https://github.com/tikv/tikv/issues/10813)